pkgs/development/compilers/ghc/ghc-9.4-PPC-Support-ELF-v2-on-powerpc64-big-endian.patch

```diff
@@ -0,0 +1,216 @@
+From 6c3205cf3d16d35396984d5fcf55c97ea1d11194 Mon Sep 17 00:00:00 2001
+From: OPNA2608 <opna2608@protonmail.com>
+Date: Thu, 11 Sep 2025 18:48:51 +0200
+Subject: [PATCH 1/2] [9.4] PPC: Support ELF v2 on powerpc64 big-endian
+
+Backport of ead75532c9dc915bfa9ebaef0ef5d148e793cc0a to 9.4.8
+---
+ compiler/GHC/CmmToAsm/PPC/CodeGen.hs      | 10 ++++++++--
+ libraries/ghc-boot/GHC/Platform/ArchOS.hs |  3 +--
+ m4/fptools_set_haskell_platform_vars.m4   |  8 +++-----
+ m4/ghc_get_power_abi.m4                   | 19 +++++++++++++++++++
+ rts/AdjustorAsm.S                         | 12 +++++-------
+ rts/StgCRun.c                             |  2 +-
+ rts/StgCRunAsm.S                          | 12 ++++++------
+ rts/adjustor/NativePowerPC.c              |  6 ++----
+ 8 files changed, 45 insertions(+), 27 deletions(-)
+ create mode 100644 m4/ghc_get_power_abi.m4
+
+diff --git a/compiler/GHC/CmmToAsm/PPC/CodeGen.hs b/compiler/GHC/CmmToAsm/PPC/CodeGen.hs
+index 7329b8c4b1..3a18eb7600 100644
+--- a/compiler/GHC/CmmToAsm/PPC/CodeGen.hs
++++ b/compiler/GHC/CmmToAsm/PPC/CodeGen.hs
+@@ -1895,8 +1895,14 @@ genCCall' config gcp target dest_regs args
+                          -- "Single precision floating point values
+                          -- are mapped to the second word in a single
+                          -- doubleword"
+-                         GCP64ELF 1      -> stackOffset' + 4
+-                         _               -> stackOffset'
++                         GCP64ELF 1 -> stackOffset' + 4
++                         -- ELF v2 ABI Revision 1.5 Section 2.2.3.3. requires
++                         -- a single-precision floating-point value
++                         -- to be mapped to the least-significant
++                         -- word in a single doubleword.
++                         GCP64ELF 2 -> case platformByteOrder platform of
++                                       BigEndian    -> stackOffset' + 4
++                                       LittleEndian -> stackOffset'
```

Comment:
Isn't this missing the `_               -> stackOffset'` line?

This seems to be still preserved in the fetched patch, why not here?

Reply:
Should be fixed now. Sorry!